.ocamlformat

```diff
@@ -0,0 +1,9 @@
+profile=ocamlformat
+version=0.14.1
+indicate-multiline-delimiters=closing-on-separate-line
+if-then-else=fit-or-vertical
+dock-collection-brackets=true
+break-struct=natural
+break-separators=before
+break-infix=fit-or-vertical
+break-infix-before-func=false
```

Comment:
can we wget this config from some central location (e.g. xs-opam), so we don't have to maintain duplicate copies across several repositories?

Reply:
I'm not sure what process do you want to automate with wget. Could you be more specific?

Updating the config file is not something that's going to be done frequently, and having it in version control is useful when we have more than one branch to maintain.

I'm not opposed to having the file in xs-opam, maybe even as a script that moves files around local repositories; but I'm not so keen on getting it removed from the version-controlled repositories.

Reply:
I'd keep the file here but also `curl` it from a central place and diff it - can be all done in the Makefile. I don't think we need to do this now but should consider it.

Reply:
> I'm not sure what process do you want to automate with wget. Could you be more specific?

specifically:

```
# xcp-idl makefile
format:
  if file_not_exist ".ocamlformat" then
    wget "https://github.com/xapi-project/xs-opam/.../.ocamlformat" -o .ocamlformat
  run_formatting

# xcp-idl .gitignore
.ocamlformat
```

> I'm not opposed to having the file in xs-opam, maybe even as a script that moves files around local repositories; but I'm not so keen on getting it removed from the version-controlled repositories.

it would still in version control, just in xs-opam instead

> Updating the config file is not something that's going to be done frequently, and having it in version control is useful when we have more than one branch to maintain.

i didn't think of that, i suppose perhaps we should leave it as it currently is